ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -61,13 +61,16 @@ object EndpointsCompanion {
                     Unauthorized("ledgerId missing in access token"))
                   applicationId <- payload.applicationId.toRightDisjunction(
                     Unauthorized("applicationId missing in access token"))
-                  party <- payload.party.toRightDisjunction(
-                    Unauthorized("party missing or not unique in access token"))
+                  actAs <- payload.actAs match {
+                    case p :: Nil => \/-(p)
```

Comment:
```suggestion
                    case Seq(p) => \/-(p)
```

Matching on `::` is quite dangerous in Scala.

Reply:
Why? We are working with `List` here, shouldn’t `::` be fine? Happy to make the change, I’d just like to understand why this is better.

Reply:
As long as the selector is `List` then it is fine. But it is very easy to change the selector's type to `Seq`, and `::` will continue to compile without warning, so the change will almost certainly go unnoticed, especially if it is separate from the type declaration as here. `::` will work for any case where the `Seq` happens to be a `List`, but if any code path uses `Vector` or anything else the `::` will simply not match.

Reply:
Makes sense, thanks for the explanation!